core/chains/evm/gas/arbitrum_estimator.go

```diff
@@ -52,20 +43,25 @@ type arbitrumEstimator struct {
 	chInitialised  chan struct{}
 	chStop         services.StopChan
 	chDone         chan struct{}
+
+	l1Oracle rollups.ArbL1GasOracle
 }
 
-func NewArbitrumEstimator(lggr logger.Logger, cfg ArbConfig, rpcClient rpcClient, ethClient ethClient) EvmEstimator {
+func NewArbitrumEstimator(lggr logger.Logger, cfg ArbConfig, ethClient feeEstimatorClient) EvmEstimator {
 	lggr = logger.Named(lggr, "ArbitrumEstimator")
+
+	l1Oracle := rollups.NewArbitrumL1GasOracle(lggr, ethClient)
+
 	return &arbitrumEstimator{
 		cfg:            cfg,
-		EvmEstimator:   NewSuggestedPriceEstimator(lggr, rpcClient, cfg),
-		client:         ethClient,
+		EvmEstimator:   NewSuggestedPriceEstimator(lggr, ethClient, cfg, config.ChainArbitrum),
```

Comment:
One problem I see with creating the l1Oracles inside the estimators instead of injecting them is that in this case you will create a new L1Oracle inside the suggested price estimator and eventually you will have two arbitrum oracles making RPC requests. It probably won't affect the functionality, but you do end up having two services in the background doing the exact same thing.